scad/switch.scad

```diff
@@ -24,185 +24,182 @@ module switch_socket_base(borders=[1,1,1,1]) {
 }
 
 module switch_socket_cutout(borders=[1,1,1,1], rotate_column=false) {
-    if (switch_type == "mx") {
-        if (use_folded_contact) {
-            mx_improved_socket_cutout(borders, rotate_column);
-        } else {
-            mx_socket_cutout(borders, rotate_column);
-        }
-    } else if (switch_type == "choc") {
-        choc_socket_cutout(borders, rotate_column);
-    } else {
-        assert(false, "switch_type is invalid");
-    }
-}
+    // Pin positions within socket cutout
+    top_pin_xy =
+        switch_type == "mx" ? [2*grid,4*grid]
+        : switch_type == "choc" ? [0,5.9]
+        : undef;
+    bottom_pin_xy =
+        switch_type == "mx" ? [-3*grid,2*grid]  // No mx bottom pin
+        : switch_type == "choc" ? [5,3.8]
+        : undef;
+    // Side pin is +-
+    side_pin_x =
+        switch_type == "mx" ? 4
+        : switch_type == "choc" ? 5.5
+        : undef;
+    diode_cutout_xy =
+        switch_type == "mx" ? [3*grid,-4*grid]
+        : switch_type == "choc" ? [-3.125, -3.8]
+        : undef;
+    col_channel_xy =
+        switch_type == "mx" ? [3*grid,-4*grid]
+        : switch_type == "choc" ? [-3.125,-3.8]
+        : undef;
+    row_channel_y =
+        switch_type == "mx" ? 4*grid
+        : switch_type == "choc" ? 5.9
+        : undef;
+    central_pin_r =
+        switch_type == "mx" ? 2.1
+        : switch_type == "choc" ? 1.75
+        : undef;
+    top_pin_cutout_r = 1;
 
-module mx_improved_socket_cutout(borders=[1,1,1,1], rotate_column=false) {
     render() translate([h_unit/2,-v_unit/2,0]) rotate([0,0,switch_rotation])
         intersection() {
             union() {
+                // Top switch pin
+                translate(top_pin_xy)
+                    cylinder(h=pcb_thickness+1,r=top_pin_cutout_r);
+
                 // Central pin
                 translate([0,0,pcb_thickness/2-socket_depth])
-                    cylinder(h=pcb_thickness+1,r=2.1);
+                    cylinder(h=pcb_thickness+1,r=central_pin_r);
+
                 // Side pins
-                for (x = [-4,4]) {
-                    translate([x*grid,0,pcb_thickness/2-socket_depth])
-                        cylinder(h=pcb_thickness+1,r=1.05);
+                if (five_pin_switch){
+                    for (x = [-side_pin_x, side_pin_x]) {
+                        translate([x*grid,0,pcb_thickness/2-socket_depth])
```

Comment:
the use of `grid` here causes the holes to not generate for chocs. I guess this makes more sense if you make the grid size `1` for chocs instead of `0`.

Reply:
Oops, missed this in testing. Thanks

Reply:
actually the more I think about it, the more I think `side_pin_x` should always be in `mm` to match the rest of the parameters, so this module won't depend on `grid` at all

Reply:
Should be able to work that myself